src/components/organism/StepBToC.js

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from 'react'
+import { injectStripe, CardElement } from 'react-stripe-elements'
+
+import { config } from '../../config'
+
+import StepsHeader from '../atom/StepsHeader'
+import InsertDetails from '../molecules/InsertDetails'
+// import MakePayment from '../molecules/MakePayment'
+// import StepAToB from './StepAToB'
+
+const cardStyle = {
+  base: {
+    color: 'black',
+    '::placeholder': {
+      color: 'grey'
+    },
+    iconColor: 'black',
+  }
+}
+
+const postCharge = payload => {
+  console.log('Posting charge: ', payload.amount)
+  return fetch(config.chargeProcessor, {
+    body: JSON.stringify(payload),
+    headers: {
+      'content-type': 'application/json'
+    },
+    method: 'POST'
+  })
+    .then(response => {
+      console.log('Success: ', response)
+    })
+    .catch(error => console.error('Error:', error))
+}
+
+class StepBToC extends Component {
+  constructor() {
+    super()
+    this.state = {
+      isHidden: true,
+      // showAmount: false
+    }
+  }
+  showPreviousStep(e) {
+    e.preventDefault();
+    this.setState({
+      showAmount: !this.state.showAmount
+    })
+  }
+  showNextStep(e) {
+    e.preventDefault();
+    this.setState({
+      isHidden: !this.state.isHidden
+    })
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault()
+    if (this.props.stripe) {
+      const name = this.state.name
+      this.props.stripe.createToken({ name })
+      .then((payload) => {
+        console.log('[token]', payload)
+        return postCharge(Object.assign(payload, this.state))
+      })
+      .then(response => { 
+        alert('Thank you for your contribution - we have sent you a confirmation email')
+        this.setState({
+          name: '',
+          email: '',
+          amount: null
+        })
+      })
+      .catch(error => console.error('Error:', error))
+      } else {
+        console.log("Stripe.js hasn't loaded yet.");
+      }
+  }
+
+  render () {
+    return (
+      <div className='cardBody'>
+         {this.state.isHidden && 
+         <fieldset id="insertDetails"> 
+          <StepsHeader stepTwo='boldStep' />
+          <form form id='contributeForm' className='form' onSubmit={this.handleSubmit}>
+            <InsertDetails />
```

Comment:
My assumption for how this would need to work is that the state would need to be held in either this organism or in one of the wrapping components. And that we would then be passing down state update functions to the InsertDetails for that component to use to update the state in the outer component. However I may be a bit out of touch with how the latest version of react manages state - ie this.state.isHiddoen.

do you have a clear idea of how the state management is working?